File: hackhive-backend/src/main/java/com/hackhive/student/service/StudentPrivacyPreferenceService.java
package com.hackhive.student.service;

import com.hackhive.student.dto.request.UpdateStudentPrivacyPreferenceRequest;
import com.hackhive.student.dto.response.StudentPrivacyPreferenceResponse;
import com.hackhive.student.entity.StudentPrivacyPreference;
import com.hackhive.student.entity.StudentProfile;

public interface StudentPrivacyPreferenceService {

    StudentPrivacyPreferenceResponse getPreferences();

    StudentPrivacyPreferenceResponse updatePreferences(UpdateStudentPrivacyPreferenceRequest request);

    StudentPrivacyPreference createDefaultPreferences(StudentProfile studentProfile);
}

File: hackhive-frontend/src/services/studentPrivacyPreferenceService.js
import api from "../api/axios";

export const studentPrivacyPreferenceService = {
    getPreferences: async () => {
        const response = await api.get("/student/privacy-preferences");
        return response.data?.data ?? response.data;
    },

    updatePreferences: async (payload) => {
        const response = await api.put("/student/privacy-preferences", payload);
        return response.data?.data ?? response.data;
    },
};

export default studentPrivacyPreferenceService;
